### PR TITLE
TestSuite: debug API

### PR DESCRIPTION
**Pull request by Sadaf-A**
## Description
This PR adds additional testcases for debug API for a better test coverage

## Changes proposed
Following testcases have been added
### debug API
log()
- [x] logs an info message
- [x] logs a warning message
- [x] logs a debug message
- [x] logs a message with special characters
- [x] handles empty log messages gracefully

## How to test it

 - Run specs/tests

## Next steps

None.

## Deploy notes

None.

### CODE REVIEW

File: spec/debug.spec.js [hunk 1]
```diff
@@ -5,21 +5,76 @@ const runner = require('./runner');
...
             assert.ok(runner.getOutput().includes('Hello! This is a simple message'));
         });
 
+        it('logs an info message', async () => {
+            runner.run(`
+                await Neutralino.debug.log('Hello! This is a info message', 'INFO');
+                let logFileContent = await Neutralino.filesystem.readFile(NL_PATH + '/neutralinojs.log');
+                await __close(logFileContent);
+            `);
+            assert.ok(runner.getOutput().includes('Hello! This is a info message'));
```
Comment (shalithasuranga) [inline spec/debug.spec.js:22]: The `INFO` option outputs the `INFO` word into the logfile, so can we add another assertion for it as well?

File: spec/debug.spec.js [hunk 2]
```diff
@@ -5,21 +5,76 @@ const runner = require('./runner');
...
             assert.ok(runner.getOutput().includes('Hello! This is an error message'));
         });
+
+        it('logs a warning message', async () => {
+            runner.run(`
+                await Neutralino.debug.log('Hello! This is a warning message', 'WARNING');
+                let logFileContent = await Neutralino.filesystem.readFile(NL_PATH + '/neutralinojs.log');
+                await __close(logFileContent);
+            `);
+            assert.ok(runner.getOutput().includes('Hello! This is a warning message'));
```
Comment (shalithasuranga) [inline spec/debug.spec.js:40]: The `WARNING` option outputs the `WARNING` word into the logfile, so can we add another assertion for it as well?

File: spec/debug.spec.js [hunk 3]
```diff
@@ -5,21 +5,76 @@ const runner = require('./runner');
...
+        });
+
+        it('logs a debug message', async () => {
+            runner.run(`
+                await Neutralino.debug.log('Hello! This is a debug message', 'DEBUG');
+                let logFileContent = await Neutralino.filesystem.readFile(NL_PATH + '/neutralinojs.log');
+                await __close(logFileContent);
+            `);
+            assert.ok(runner.getOutput().includes('Hello! This is a debug message'));
```
Comment (shalithasuranga) [inline spec/debug.spec.js:49]: The `DEBUG` option outputs the `DEBUG` word into the logfile, so can we add another assertion for it as well?

**Review by shalithasuranga — CHANGES_REQUESTED** [review]
Thanks for sending this PR @Sadaf-A. I've added some comments to several changes. We can merge this after solving those requested changes mentioned in comments :tada:

**Comment by Sadaf-A** [comment]
> Thanks for sending this PR @Sadaf-A. I've added some comments to several changes. We can merge this after resolving those requested changes mentioned in comments :tada:

Sure, I'll resolve the comments by the EOD. Thank you for the review!

**Review by shalithasuranga — APPROVED** [review]
Thanks, looks great now :tada:

**Pull request merged**